scripts/perfmon

```diff
@@ -379,8 +381,15 @@ def get_percent_mem_usage(ignored):
         memfd = open('/proc/meminfo', 'r')
         memlist = memfd.readlines()
         memfd.close()
-        memdict = [ m.split(':', 1) for m in memlist ]
-        memdict = dict([(k.strip(), float(re.search(r'\d+', v.strip()).group(0))) for (k,v) in memdict])
+        memorylist = [ m.split(':', 1) for m in memlist ]
+        memdict = {}
+        for item in memorylist:
+            memtype = item[0].strip()
+            size = item[1].strip()
+            match = re.search(r'\d+', size.strip())
+            assert match
```

Comment:
If there is no match this will raise an assertion error, it might be better to raise an error that gives you more information on what went wrong: in this case including the original `size` that we tried to match with the regex might be useful, so we see what it contains instead of digits.

Reply:
- This change is not covered by tests.
  When installing and configuring the Codecov plugin for Chrome- or Firefox-based browsers, a red bar is shown in front of these lines. I recommend submitters to install and configure it (more info later, I'm in a meeting now) or check the code coverage on the Link given by the Codecov.io
- We also need to run mypy checks
- Since there are so many unresolved comments, and many changes, I think it's better to revert the Python3 changes as a fix, and move this to the Python3 branch instead to give this more time for review.

Reply:
This is also for passing the pytype check.

Reply:
Agree with @edwintorok.
I'm assuming your modification is inspired by @bernhardkaindl's draft PR (https://github.com/xapi-project/xen-api/pull/5365/files#r1458703889) to avoid pytype errors. However, it would be preferable to make more elegant adjustments to display insightful exceptions.